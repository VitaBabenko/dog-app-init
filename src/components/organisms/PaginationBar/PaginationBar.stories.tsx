import type { Meta, StoryObj } from '@storybook/react';
import { PaginationBar } from '.';

const meta = {
  title: 'Organisms/PaginationBar',
  component: PaginationBar
} satisfies Meta<typeof PaginationBar>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    page: 1
  }
};
