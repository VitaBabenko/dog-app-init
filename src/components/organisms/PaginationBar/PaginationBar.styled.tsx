import { Pagination, styled } from '@mui/material';

export const StyledPagination = styled(Pagination)({
  display: 'flex',
  justifyContent: 'flex-end'
});
