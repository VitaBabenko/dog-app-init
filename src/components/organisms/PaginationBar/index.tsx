import { Stack } from '@mui/material';
import { FC } from 'react';
import { useGetImagesWithFavourites } from '../../../hooks';
import { StyledPagination } from './PaginationBar.styled';

type PaginationBarProps = {
  page: number;
  handleChange: (_event: React.ChangeEvent<unknown>, value: number) => void;
};

export const PaginationBar: FC<PaginationBarProps> = ({
  page,
  handleChange
}) => {
  const { isLoading } = useGetImagesWithFavourites({ page });

  return (
    <Stack>
      {!isLoading && (
        <StyledPagination
          count={28}
          page={page}
          onChange={handleChange}
          showFirstButton
          showLastButton
        />
      )}
    </Stack>
  );
};
