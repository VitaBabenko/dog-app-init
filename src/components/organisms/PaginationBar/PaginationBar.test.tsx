import { composeStories } from '@storybook/react';
import * as stories from './PaginationBar.stories';
import { testSnapshot } from '../../../test/utils/snapshot.utils';

describe('PaginationBar', () => {
  const { Primary } = composeStories(stories);
  testSnapshot(<Primary />);
});
