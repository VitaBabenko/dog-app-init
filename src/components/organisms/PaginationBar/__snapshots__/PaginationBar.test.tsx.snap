// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationBar should match snapshot Component 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div>
  <div
    class="MuiStack-root emotion-0"
  />
</div>
`;
